In [1]:
import pandas as pd
import numpy as np
df=pd.read_csv('정상적 리뷰여행지.csv')
df

여행지 일치여부
0           경복궁    O
1          삼청동길    O
2          인사동길    O
3           조계사    O
4           청와대    O
..          ...  ...
608     알뜨르 비행장    O
609     이중섭 미술관    O
610         쇠소깍    O
611  휴애리 자연생활공원    O
612    제주 감귤박물관    O

[613 rows x 2 columns]

In [2]:
df_t=df[['여행지']]
df_t.head()

여행지
0   경복궁
1  삼청동길
2  인사동길
3   조계사
4   청와대

In [3]:
# 크롤링 라이브러리 불러오기
import pandas as pd
import numpy as np
from selenium import webdriver
import time
from bs4 import BeautifulSoup
import requests
from selenium.webdriver.common.keys import Keys
import urllib
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.common.keys import Keys
from selenium.webdriver.support.ui import WebDriverWait
from selenium.webdriver.support import expected_conditions as EC
from selenium.webdriver.support.ui import Select,WebDriverWait

#place_main_ct > div > div > div > div.ct_box_area > div.bizinfo_area > div > div:nth-child(2) > div > ul > li:nth-child(1) > span > a > span.txt

#place_main_ct > div > div > div > div.ct_box_area > div.bizinfo_area > div > div:nth-child(1) > div > ul > li:nth-child(1) > span > a > span.txt

In [123]:
options=webdriver.ChromeOptions()
options.add_argument('user_agent=Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/101.0.4951.67 Safari/537.36')
options.add_experimental_option("excludeSwitches", ["enable-logging"])

basepath='C:/Users/이신행/OneDrive/바탕 화면/chromedriver_win32(2)/'
driver=webdriver.Chrome()
url='https://www.tripadvisor.co.kr/'
driver.get(url)
address=[]
for i in range(len(df_t)):
    search=df_t.여행지[i]
    driver.find_element_by_css_selector('#lithium-root > main > div.dqRmR.dcDXR.dJjeH > div > div > div.weiIG.Z0.Wh.fRhqZ > form > input.fhEMT._G.B-.z._J.Cj.R0').send_keys(search)
    driver.find_element_by_css_selector('#lithium-root > main > div.dqRmR.dcDXR.dJjeH > div > div > div.weiIG.Z0.Wh.fRhqZ > form > input.fhEMT._G.B-.z._J.Cj.R0').send_keys(Keys.ENTER)
    time.sleep(1)

    soup=BeautifulSoup(driver.page_source,'html.parser')
    time.sleep(1)
    mydata=soup.find('div',attrs={'class':'mobile-address-text'})
#     address.append(mydata.get_text().strip())
    print(mydata)
    address.append(mydata)
    driver.back()
    time.sleep(1)

<div class="mobile-address-text">종로구 사직로 161, 서울</div>
None
<div class="mobile-address-text">종로구 인사동, 서울</div>
<div class="mobile-address-text">종로구 견지동 45번지, 서울</div>
<div class="mobile-address-text">종로구 청와대로 1, 서울</div>
<div class="mobile-address-text">종로구 종로 54, 서울</div>
<div class="mobile-address-text">종로구 삼청로 37 , 서울</div>
None
<div class="mobile-address-text">종로구 새문안로 29, 서울</div>
None
<div class="mobile-address-text">종로구, 서울</div>
<div class="mobile-address-text">종로구 종로 157, 서울</div>
<div class="mobile-address-text">종로구 신문로2가 2-1, 서울</div>
<div class="mobile-address-text">종로구 창경궁로 185, 서울</div>
<div class="mobile-address-text">종로구 율곡로 99, 서울</div>
<div class="mobile-address-text">종로구 계동길 37, 서울</div>
<div class="mobile-address-text">중구, 서울</div>
None
None
<div class="mobile-address-text">중구 명동길 74, 서울</div>
<div class="mobile-address-text">서울</div>
None
<div class="mobile-address-text">중구 남대문시장4길 21, 서울</div>
<div class="mobile-address-text">중구 태평로1가, 서울</div>
<div class="mobile-

WebDriverException: Message: target frame detached
  (Session info: chrome=101.0.4951.67)


In [4]:
options=webdriver.ChromeOptions()
options.add_argument('user_agent=Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/101.0.4951.67 Safari/537.36')
options.add_experimental_option("excludeSwitches", ["enable-logging"])

basepath='C:/Users/이신행/OneDrive/바탕 화면/chromedriver_win32(2)/'
driver=webdriver.Chrome()
url='https://www.tripadvisor.co.kr/'
driver.get(url)
address=[]
for i in range(len(df_t)):
    try:
        search=df_t.여행지[i]
        driver.find_element_by_css_selector('#lithium-root > main > div.dqRmR.dcDXR.dJjeH > div > div > div.weiIG.Z0.Wh.fRhqZ > form > input.fhEMT._G.B-.z._J.Cj.R0').send_keys(search)
        driver.find_element_by_css_selector('#lithium-root > main > div.dqRmR.dcDXR.dJjeH > div > div > div.weiIG.Z0.Wh.fRhqZ > form > input.fhEMT._G.B-.z._J.Cj.R0').send_keys(Keys.ENTER)
        time.sleep(1)

        soup=BeautifulSoup(driver.page_source,'html.parser')
        mydata=soup.find('div',attrs={'class':'mobile-address-text'})
        
        if mydata==None:
            mydata=soup.find('div',atters={'class':'address-text'})
            if mydata==None:
                address.append('주소가 나오지 않음')
                driver.back()
                time.sleep()
            else:
                print(mydata)
                address.append(mydata)
                driver.back()
                time.sleep(1)
        else:
            print(mydata)
            address.append(mydata)
            driver.back()
            time.sleep(1)
    except:
        address.append('주소가 명확하지 않음')
        pass

<div class="mobile-address-text">종로구 사직로 161, 서울</div>
<div class="mobile-address-text">107 종로구 삼청동, 서울</div>
<div class="mobile-address-text">종로구 인사동, 서울</div>
<div class="mobile-address-text">종로구 견지동 45번지, 서울</div>
<div class="mobile-address-text">종로구 청와대로 1, 서울</div>
<div class="mobile-address-text">종로구 종로 54, 서울</div>
<div class="mobile-address-text">종로구 삼청로 37 , 서울</div>
<div class="mobile-address-text">사직동 1-48, 서울</div>
<div class="mobile-address-text">종로구 새문안로 29, 서울</div>
<div class="mobile-address-text">종로구 종로2가, 서울</div>
<div class="mobile-address-text">종로구, 서울</div>
<div class="mobile-address-text">종로구 종로 157, 서울</div>
<div class="mobile-address-text">종로구 신문로2가 2-1, 서울</div>
<div class="mobile-address-text">종로구 창경궁로 185, 서울</div>
<div class="mobile-address-text">종로구 율곡로 99, 서울</div>
<div class="mobile-address-text">종로구 계동길 37, 서울</div>
<div class="mobile-address-text">중구, 서울</div>
<div class="mobile-address-text">종로구 삼일대로 464, 서울</div>
<div class="mobile-address-text">종로구 동

<div class="mobile-address-text">서원면 경강로유현1길 30, 횡성</div>
<div class="mobile-address-text">안흥면 안흥로 27, 횡성</div>
<div class="mobile-address-text">갑천면 어답산로, 횡성</div>
<div class="mobile-address-text">김삿갓면 김삿갓로 432-10, 영월</div>
<div class="mobile-address-text">영월읍 단종로190, 영월</div>
<div class="mobile-address-text">영월읍 청령포로 133, 영월</div>
<div class="mobile-address-text">무릉도원면 무릉법흥로 1352, 영월</div>
<div class="mobile-address-text">영월읍 천문대길 397, 영월</div>
<div class="mobile-address-text">영월읍 동강로 716, 영월</div>
<div class="mobile-address-text">봉평면 흥정리, 평창</div>
<div class="mobile-address-text">대관령면 꽃밭양지길 708-9, 평창</div>
<div class="mobile-address-text">대관령면 대관령마루길 483-32, 평창</div>
<div class="mobile-address-text">진부면 오대산로 374-8, 평창</div>
<div class="mobile-address-text">평창군 진부면 오대산로 1211-14, 평창</div>
<div class="mobile-address-text">대관령면 비안길 159-4, 평창</div>
<div class="mobile-address-text">미탄면 청옥산길 583-76, 평창</div>
<div class="mobile-address-text">봉평면 무이리 58, 평창</div>
<div class="mobile-address-te

<div class="mobile-address-text">고창</div>
<div class="mobile-address-text">관산읍 옥당리 25-1, 장흥</div>
<div class="mobile-address-text">유치면 보림사로 224, 장흥</div>
<div class="mobile-address-text">68 - 35 Dasan-dang Road, 강진</div>
<div class="mobile-address-text">강진읍 영랑생가길 15, 강진</div>
<div class="mobile-address-text">저동, 강릉</div>
<div class="mobile-address-text">성전면 무위사로 308, 강진</div>
<div class="mobile-address-text">마량면 마량3길 46-2, 강진</div>
<div class="mobile-address-text">송지면 땅끝해안로, 해남</div>
<div class="mobile-address-text">삼산면 대흥사길 400, 해남</div>
<div class="mobile-address-text">삼산면 대흥사길 88-45, 해남</div>
<div class="mobile-address-text">247, Seojeong-ri, Songji-myeon, 해남</div>
<div class="mobile-address-text">San 18, Donggurim-ri, Gunseo-myeon, 영암</div>
<div class="mobile-address-text">1153-2 Suho-ri, Hampyeong-eup, 함평</div>
<div class="mobile-address-text">백수읍 해안로, 영광</div>
<div class="mobile-address-text">8, Moak-ri, Bulgap-myeon, 영광</div>
<div class="mobile-address-text">26, Yaksu-ri, Bukha-

<div class="mobile-address-text">사하구 낙동남로 1240, 부산</div>
<div class="mobile-address-text">금정구 청룡동 546, 부산</div>
<div class="mobile-address-text">부산</div>
<div class="mobile-address-text">기장군 기장읍 용궁길 86, 부산</div>
<div class="mobile-address-text">기장읍 읍내로104번길 16, 부산</div>
<div class="mobile-address-text">기장군 장안읍 장안로 482, 부산</div>
<div class="mobile-address-text">남구 장생포고래로244, 울산</div>
<div class="mobile-address-text">동구 동해안로 653 (주전동), 울산</div>
<div class="mobile-address-text">상북면 자수정로 212, 울산</div>
<div class="mobile-address-text">울주군 서생면 대송리, 울산</div>
<div class="mobile-address-text">두동면 반구대안길 254, 울산</div>
<div class="mobile-address-text">중구 서성로 10, 대구</div>
<div class="mobile-address-text">조천읍 함덕리, 제주</div>
<div class="mobile-address-text">구좌읍 종달리 산28, 제주</div>
<div class="mobile-address-text">용두암길 15, 제주</div>
<div class="mobile-address-text">삼성로 40, 제주</div>
<div class="mobile-address-text">한경면 저지리 예술인마을, 제주</div>
<div class="mobile-address-text">조천읍 북촌리 976, 제주</div>
<div class="m

<div class="address" onclick="widgetEvCall('handlers.openResult', event, this, '/Attraction_Review-g294197-d324888-Reviews-Gyeongbokgung_Palace-Seoul.html', {type: 'ATTRACTIONS',element: this,index: 0,section: 0,locationId: '324888',parentId: '294197',elementType: 'address',selectedId: '324888'});"><div class="address-text">종로구 사직로 161, 서울, 대한민국</div><div class="mobile-address-text">종로구 사직로 161, 서울</div></div>

<div class="mobile-address-text">107 종로구 삼청동, 서울</div>



In [142]:
len(address)

613

In [141]:
options=webdriver.ChromeOptions()
options.add_argument('user_agent=Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/101.0.4951.67 Safari/537.36')
options.add_experimental_option("excludeSwitches", ["enable-logging"])

basepath='C:/Users/이신행/OneDrive/바탕 화면/chromedriver_win32(2)/'
driver=webdriver.Chrome()
url='https://www.tripadvisor.co.kr/'
driver.get(url)

for i in range(,len(df_t)):
    try:
        search=df_t.여행지[i]
        driver.find_element_by_css_selector('#lithium-root > main > div.dqRmR.dcDXR.dJjeH > div > div > div.weiIG.Z0.Wh.fRhqZ > form > input.fhEMT._G.B-.z._J.Cj.R0').send_keys(search)
        driver.find_element_by_css_selector('#lithium-root > main > div.dqRmR.dcDXR.dJjeH > div > div > div.weiIG.Z0.Wh.fRhqZ > form > input.fhEMT._G.B-.z._J.Cj.R0').send_keys(Keys.ENTER)
        time.sleep(1)

        soup=BeautifulSoup(driver.page_source,'html.parser')
        mydata=soup.find('div',attrs={'class':'mobile-address-text'})
        
        if mydata==None:
            mydata=soup.find('div',atters={'class':'address-text'})
            if mydata==None:
                address.append('주소가 나오지 않음')
                driver.back()
                time.sleep()
            else:
                print(mydata)
                address.append(mydata)
                driver.back()
                time.sleep(1)
        else:
            print(mydata)
            address.append(mydata)
            driver.back()
            time.sleep(1)
    except:
        address.append('주소가 명확하지 않음')
        pass

<div class="mobile-address-text">대구면 청자촌길 21-11, 강진</div>
<div class="mobile-address-text">마량면 마량3길 46-2, 강진</div>
<div class="mobile-address-text">송지면 땅끝해안로, 해남</div>
None
None
None
<div class="mobile-address-text">247, Seojeong-ri, Songji-myeon, 해남</div>
None
<div class="mobile-address-text">San 18, Donggurim-ri, Gunseo-myeon, 영암</div>
<div class="mobile-address-text">1153-2 Suho-ri, Hampyeong-eup, 함평</div>
<div class="mobile-address-text">백수읍 해안로, 영광</div>
None
<div class="mobile-address-text">8, Moak-ri, Bulgap-myeon, 영광</div>
None
<div class="mobile-address-text">신지면 명사십리길 85-105, 완도</div>
<div class="mobile-address-text">64, Sacheon-ri, 진도</div>
<div class="mobile-address-text">208, Unsu-ri, Hwagae-myeon, 하동</div>
<div class="mobile-address-text">군내면 만금길 112-41, 진도</div>
<div class="mobile-address-text">Cheolmagil 25, Jindo eup, 진도</div>
<div class="mobile-address-text">임회면 하미길 39, 진도</div>
<div class="mobile-address-text">신안군, 신안</div>
<div class="mobile-address-text">Dadohae Na

None
<div class="mobile-address-text">한림읍 협재리 2497-1, 제주</div>
<div class="mobile-address-text">한림읍 금능길 119-10, 제주</div>
None
<div class="mobile-address-text">상효동 산220-1, 서귀포</div>
<div class="mobile-address-text">제주도</div>
<div class="mobile-address-text">한경면 녹차분재로 675, 제주</div>
None
<div class="mobile-address-text">삼성로 22, 제주</div>
<div class="mobile-address-text">관덕로 19, 제주</div>
None
<div class="mobile-address-text">좌읍 송당리 산164-1, 제주</div>
<div class="mobile-address-text">한림읍 한림로 300, 제주</div>
<div class="mobile-address-text">성산읍 신양리, 서귀포</div>
<div class="mobile-address-text">성산읍 삼달리 437-5, 서귀포</div>
<div class="mobile-address-text">성산읍 삼달리 437-5, 서귀포</div>
<div class="mobile-address-text">법환상로2번길 97-13, 서귀포</div>
<div class="mobile-address-text">성산읍 성산리 104, 서귀포</div>
<div class="mobile-address-text">중문관광로110번길 31, 서귀포</div>
None
<div class="mobile-address-text">천지동, 서귀포</div>
<div class="mobile-address-text">이어도로 36-30, 서귀포</div>
<div class="mobile-address-text">동홍동, 서귀포</div>
<

<div class="address" onclick="widgetEvCall('handlers.openResult', event, this, '/Attraction_Review-g294197-d324888-Reviews-Gyeongbokgung_Palace-Seoul.html', {type: 'ATTRACTIONS',element: this,index: 0,section: 0,locationId: '324888',parentId: '294197',elementType: 'address',selectedId: '324888'});"><div class="address-text">종로구 사직로 161, 서울, 대한민국</div><div class="mobile-address-text">종로구 사직로 161, 서울</div></div>

In [107]:
address

[<div class="mobile-address-text">종로구 사직로 161, 서울</div>,
 None,
 None,
 None,
 <div class="mobile-address-text">종로구 청와대로 1, 서울</div>,
 <div class="mobile-address-text">종로구 종로 54, 서울</div>,
 <div class="mobile-address-text">종로구 삼청로 37 , 서울</div>,
 <div class="mobile-address-text">사직동 1-48, 서울</div>,
 None,
 <div class="mobile-address-text">종로구 종로2가, 서울</div>,
 None,
 <div class="mobile-address-text">종로구 종로 157, 서울</div>,
 None,
 None,
 <div class="mobile-address-text">종로구 율곡로 99, 서울</div>,
 <div class="mobile-address-text">종로구 계동길 37, 서울</div>,
 <div class="mobile-address-text">중구, 서울</div>,
 <div class="mobile-address-text">종로구 삼일대로 464, 서울</div>,
 None,
 <div class="mobile-address-text">중구 명동길 74, 서울</div>,
 <div class="mobile-address-text">서울</div>,
 <div class="mobile-address-text">중구 세종대로 99, 서울</div>,
 <div class="mobile-address-text">중구 남대문시장4길 21, 서울</div>,
 <div class="mobile-address-text">중구 태평로1가, 서울</div>,
 None,
 <div class="mobile-address-text">중구 세종대로 40, 서울</div>,
 None,

In [5]:
len(df_t)

613

In [118]:
options=webdriver.ChromeOptions()
options.add_argument('user_agent=Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/101.0.4951.67 Safari/537.36')
options.add_experimental_option("excludeSwitches", ["enable-logging"])

basepath='C:/Users/이신행/OneDrive/바탕 화면/chromedriver_win32(2)/'
driver=webdriver.Chrome()
url='https://www.tripadvisor.co.kr/'
driver.get(url)
address=[]
for i in range(len(df_t)):
    search=df_t.여행지[i]
    driver.find_element_by_css_selector('#lithium-root > main > div.dqRmR.dcDXR.dJjeH > div > div > div.weiIG.Z0.Wh.fRhqZ > form > input.fhEMT._G.B-.z._J.Cj.R0').send_keys(search)
    driver.find_element_by_css_selector('#lithium-root > main > div.dqRmR.dcDXR.dJjeH > div > div > div.weiIG.Z0.Wh.fRhqZ > form > input.fhEMT._G.B-.z._J.Cj.R0').send_keys(Keys.ENTER)
    time.sleep(1)

    soup=BeautifulSoup(driver.page_source,'html.parser')
    time.sleep(1)
    mydata=soup.find('div',attrs={'class':'mobile-address-text'})
#     address.append(mydata.get_text().strip())
    print(mydata)
    address.append(mydata)
    driver.back()
    time.sleep(1)

<div class="mobile-address-text">종로구 사직로 161, 서울</div>
None
<div class="mobile-address-text">종로구 인사동, 서울</div>
<div class="mobile-address-text">종로구 견지동 45번지, 서울</div>
<div class="mobile-address-text">종로구 청와대로 1, 서울</div>
<div class="mobile-address-text">종로구 종로 54, 서울</div>
<div class="mobile-address-text">종로구 삼청로 37 , 서울</div>


WebDriverException: Message: target frame detached
  (Session info: chrome=101.0.4951.67)


In [114]:
options=webdriver.ChromeOptions()
options.add_argument('user_agent=Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/101.0.4951.67 Safari/537.36')
options.add_experimental_option("excludeSwitches", ["enable-logging"])

basepath='C:/Users/이신행/OneDrive/바탕 화면/chromedriver_win32(2)/'
driver=webdriver.Chrome()
url='https://www.tripadvisor.co.kr/'
driver.get(url)

for i in range(57,len(df_t)):
    search=df_t.여행지[i]
    driver.find_element_by_css_selector('#lithium-root > main > div.dqRmR.dcDXR.dJjeH > div > div > div.weiIG.Z0.Wh.fRhqZ > form > input.fhEMT._G.B-.z._J.Cj.R0').send_keys(search)
    driver.find_element_by_css_selector('#lithium-root > main > div.dqRmR.dcDXR.dJjeH > div > div > div.weiIG.Z0.Wh.fRhqZ > form > input.fhEMT._G.B-.z._J.Cj.R0').send_keys(Keys.ENTER)
    time.sleep(1)

    soup=BeautifulSoup(driver.page_source,'html.parser')
    mydata=soup.find('div',attrs={'class':'mobile-address-text'})
#     address.append(mydata.get_text().strip())
    print(mydata)
    address.append(mydata)
    driver.back()
    time.sleep(1)

None
<div class="mobile-address-text">종로구 이화동, 서울</div>
<div class="mobile-address-text">서울</div>
<div class="mobile-address-text">성동구 뚝섬로 273, 서울</div>
None
None
<div class="mobile-address-text">성북구 선잠로5길 68, 서울</div>
None
<div class="mobile-address-text">도봉구 도봉산길, 서울</div>
<div class="mobile-address-text">화랑로 421, 서울</div>
<div class="mobile-address-text">서대문구 현저동 941, 서울</div>
<div class="mobile-address-text">서대문구 통일로 484, 서울</div>
<div class="mobile-address-text">마포구 양화진길 46, 서울</div>
<div class="mobile-address-text">영등포구 선유로 343, 서울</div>


WebDriverException: Message: unknown error: cannot determine loading status
from target frame detached
  (Session info: chrome=101.0.4951.67)


In [6]:
len(address)

731

In [7]:
for i in range(len(address)):
    try: 
        address[i]=address[i].get_text().strip()
    except:
        pass

In [147]:
address

['종로구 사직로 161, 서울',
 None,
 '종로구 인사동, 서울',
 '종로구 견지동 45번지, 서울',
 '종로구 청와대로 1, 서울',
 '종로구 종로 54, 서울',
 '종로구 삼청로 37 , 서울',
 None,
 '종로구 새문안로 29, 서울',
 None,
 '종로구, 서울',
 '종로구 종로 157, 서울',
 '종로구 신문로2가 2-1, 서울',
 '종로구 창경궁로 185, 서울',
 '종로구 율곡로 99, 서울',
 '종로구 계동길 37, 서울',
 '중구, 서울',
 None,
 None,
 '중구 명동길 74, 서울',
 '서울',
 None,
 '중구 남대문시장4길 21, 서울',
 '중구 태평로1가, 서울',
 '중구 남대문로 39, 서울',
 '중구 세종대로 40, 서울',
 '중구 충정로1가 75, 서울',
 '중구 퇴계로34길 28, 서울',
 '중구 마른내로 71, 서울',
 None,
 None,
 '용산구 서빙고로 137, 서울',
 '용산구 이태원로 29, 서울',
 '동대문구 회기로 57 , 서울',
 '성북구 보국문로 262, 서울',
 '종로구 이화동, 서울',
 '서울',
 '성동구 뚝섬로 273, 서울',
 '동대문구 청량리동 산1-157, 서울',
 '동대문구 약령중앙로8길 10, 서울',
 '성북구 선잠로5길 68, 서울',
 None,
 '도봉구 도봉산길, 서울',
 '화랑로 421, 서울',
 '서대문구 현저동 941, 서울',
 '서대문구 통일로 484, 서울',
 '마포구 양화진길 46, 서울',
 '영등포구 선유로 343, 서울',
 None,
 '영등포구 63로 50, 서울',
 None,
 '관악구 관악로, 서울',
 '서초구 헌인릉길 34 , 서울',
 None,
 None,
 '16-1, Neungsan-ri, Buyeo-eup, 부여',
 '송파구 올림픽로 424, 서울',
 '송파구 올림픽로 424, 서울',
 '중구 을지로 281, 서울',
 None,
 '길주로 1, 부천',
 N

In [8]:
data=pd.DataFrame(address)
data.to_csv('주소1차.csv',encoding='utf-8-sig',index=False)